### ISSUE TITLE
Wayland: alacritty ignores activated configure option, and instead relies on keyboard focus

### ISSUE THREAD

**Issue by buffet** [alacritty/alacritty#3139]
Alacritty seems to decide if it should draw the focused cursor or the unfocused one, depending on if it has keyboard focus, instead of relying on the `activated` configure option.

Additionally it starts assuming it *has* keyboard focus, making it impossible to open an alacritty window, without focus (at least visually).

It'd be nice if it would support the configure option, or at least start assuming it has no focus.

### System

OS: Linux
Version: alacritty 0.4.0
Linux: Wayland (kiwmi)

**Comment by chrisduerr** [comment]
> instead of relying on the activated configure option

What exactly is the `activated` configure option?

**Comment by buffet** [comment]
@chrisduerr It's the compositors way of telling the window if it should draw itself in a focused, or an unfocused state (https://github.com/wayland-project/wayland-protocols/blob/733de76221223c2ad3ce94a588015e88c4c9dcc7/stable/xdg-shell/xdg-shell.xml#L752)

Actually seems like the "assuming focus" part was fixed in d387ebe, but later broken again.

**Comment by chrisduerr** [comment]
Could you share the output of `alacritty --print-events | grep "Focus"`?

**Comment by buffet** [comment]
No output.

**Comment by chrisduerr** [comment]
But it does output something when you start focusing the window, right?

**Comment by buffet** [comment]
Yes, `[2020-01-01 17:39] [INFO] glutin event: WindowEvent { window_id: WindowId(Wayland(WindowId(94711898852944))), event: Focused(true) }`.

**Comment by chrisduerr** [comment]
Also, could you share the output of this command when the `activated` option is not set? At least on X11 it seems like the default state is no focus and an explicit focus event is sent at startup, to fix this without upstream changes this would need to be the case on Windows/macOS and Wayland too.

**Comment by buffet** [comment]
It prints nothing, no matter if it's set or not.

Prints the line above as soon as the window gets keyboard focus.

**Comment by buffet** [comment]
```patch
diff --git a/alacritty_terminal/src/term/mod.rs b/alacritty_terminal/src/term/mod.rs
index 4aafde1..574d29f 100644
--- a/alacritty_terminal/src/term/mod.rs
+++ b/alacritty_terminal/src/term/mod.rs
@@ -884,7 +884,7 @@ impl<T> Term<T> {
             auto_scroll: config.scrolling.auto_scroll,
             clipboard,
             event_proxy,
-            is_focused: true,
+            is_focused: false,
             title: config.window.title.clone(),
             title_stack: Vec::new(),
         }
```

This should "fix" what bd6e5a9 fixed, which might be enough, since every compositor seems to bungle the `activated` configure option with keyboard focus anyways.

**Comment by kchibisov** [comment]
@buffet This wont fix things in general tbh. We should fix this issue in our windowing library(winit) and make it send a proper Focused events on startup. While you can probably workaround the issue on Wayland with your diff, you'll break Windows (not aware of macOS here).

**Comment by chrisduerr** [comment]
> This should "fix" what bd6e5a9 fixed and 729eef0 broke, which might be enough, since every compositor seems to bungle the activated configure option with keyboard focus anyways.

You've said yourself that your compositor doesn't send focus at startup whet `activated` is disabled, so this would just break it for the majority of users. And yeah, as @kchibisov has correctly pointed out, it will also break Windows (though that could be fixed with a `cfg` annotation).

**Comment by chrisduerr** [comment]
See https://github.com/rust-windowing/winit/issues/1077 for upstream tracking.
